server/asset/eth/coiner.go

```diff
@@ -0,0 +1,248 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package eth
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"decred.org/dcrdex/server/asset"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type swapCoinerType uint8
+
+const (
+	sctInit swapCoinerType = iota
+	sctRedeem
+)
+
+var _ asset.Coin = (*swapCoiner)(nil)
+
+type swapCoiner struct {
```

Comment:
Naming on this correct? It should be the interface with `-er` and not the implementer?

Reply:
I think the interface should have the `-er` so mass renamed.